### ISSUE TITLE
hello how can i use this on 2D img conv? 

### ISSUE THREAD

**Issue by gwspotex** [AmirAlavi/sparsely-connected-keras#1]
# Block 1   
    x = Conv2D(64, (3, 3), activation='relu', padding='same', name='block1_conv1')(img_input)
    x = Conv2D(64, (3, 3), activation='relu', padding='same', name='block1_conv2')(x)
    x = MaxPooling2D((2, 2), strides=(2, 2), name='block1_pool')(x)

    # Block 2
    x = Conv2D(128, (3, 3), activation='relu', padding='same', name='block2_conv1')(x)
    x = Conv2D(128, (3, 3), activation='relu', padding='same', name='block2_conv2')(x)
    x = MaxPooling2D((2, 2), strides=(2, 2), name='block2_pool')(x)

    # Block 3
    x = Conv2D(256, (3, 3), activation='relu', padding='same', name='block3_conv1')(x)
    x = Conv2D(256, (3, 3), activation='relu', padding='same', name='block3_conv2')(x)
    x = Conv2D(256, (3, 3), activation='relu', padding='same', name='block3_conv3')(x)
    x = MaxPooling2D((2, 2), strides=(2, 2), name='block3_pool')(x)

    # Block 4
    x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv1')(x)
    x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv2')(x)
    x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv3')(x)
    x = MaxPooling2D((2, 2), strides=(2, 2), name='block4_pool')(x)

    # Block 5
    x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block5_conv1')(x)
    x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block5_conv2')(x)
    x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block5_conv3')(x)
    -------------------------------------------------------------------------------------------
     standard vgg16 out 7*7*512
     ------------------------------------------------------------------------------------------
    
    out = Flatten(name='flatten'))(x)
    out = Dense(4096, activation='relu', name='fc1'))(out)
    out =Dropout(0.5))(out)
    out = Dense(4096, activation='relu', name='fc2'))(out)
    out = Dropout(0.5))(out)

    out_class = Dense(nb_classes, activation='softmax' )(out)

    '''

**Comment by AmirAlavi** [comment]
Hi @gwspotex, sorry for the huge delay. Currently this isn't made for Convolution, and I'm not sure how one could make use of this in the context of Convolution. If you have some thoughts on how that can be done, please let me know. My initial thoughts are that it doesn't really make sense to apply this to convolutional layers, since they are already doing this in some sense: conv layers are already "sparse" in the sense that the neurons are only looking at portions of the image, and are not connected to every pixel. For a more detailed explanation of what I mean, you can check out Hugo Larochelle's great slides [here](http://info.usherbrooke.ca/hlarochelle/ift725/9_02_local_connectivity.pdf) (from module 9 of his course [here](http://info.usherbrooke.ca/hlarochelle/neural_networks/content.html)).
